client/members_test.go

```diff
@@ -520,3 +537,63 @@ func TestHTTPMembersAPIListError(t *testing.T) {
 		}
 	}
 }
+
+func TestHTTPMembersAPILeaderSuccess(t *testing.T) {
+	wantAction := &membersAPIActionLeader{}
+	mAPI := &httpMembersAPI{
+		client: &actionAssertingHTTPClient{
+			t:   t,
+			act: wantAction,
+			resp: http.Response{
+				StatusCode: http.StatusOK,
+			},
+			body: []byte(`{"id":"94088180e21eb87b","name":"node2","peerURLs":["http://127.0.0.1:7002"],"clientURLs":["http://127.0.0.1:4002"]}`),
+		},
+	}
+
+	wantResponseMember := &Member{
+		ID:         "94088180e21eb87b",
+		Name:       "node2",
+		PeerURLs:   []string{"http://127.0.0.1:7002"},
+		ClientURLs: []string{"http://127.0.0.1:4002"},
+	}
+
+	m, err := mAPI.Leader(context.Background())
+	if err != nil {
+		t.Errorf("got non-nil err: %#v", err)
+	}
+	if !reflect.DeepEqual(wantResponseMember, m) {
+		t.Errorf("incorrect Members: want=%#v got=%#v", wantResponseMember, m)
+	}
+}
+
+func TestHTTPMembersAPILeaderError(t *testing.T) {
+	tests := []httpClient{
+		// generic httpClient failure
+		&staticHTTPClient{err: errors.New("fail!")},
+
+		// unrecognized HTTP status code
+		&staticHTTPClient{
+			resp: http.Response{StatusCode: http.StatusTeapot},
+		},
+
+		// fail to unmarshal body on StatusOK
+		&staticHTTPClient{
+			resp: http.Response{
+				StatusCode: http.StatusOK,
+			},
+			body: []byte(`[{"id":"XX`),
+		},
+	}
+
+	for i, tt := range tests {
+		mAPI := &httpMembersAPI{client: tt}
+		ms, err := mAPI.Leader(context.Background())
+		if err == nil {
+			t.Errorf("#%d: got nil err", i)
+		}
+		if ms != nil {
+			t.Errorf("#%d: got non-nil Member slice", i)
```

Comment:
On the second thought, this case shouldn't assume non nil Member slice. It should assume non nil member pointer.